distribution/xfer/download.go

```diff
@@ -70,7 +86,7 @@ type DownloadDescriptor interface {
 	// before).
 	DiffID() (layer.DiffID, error)
 	// Download is called to perform the download.
-	Download(ctx context.Context, progressOutput progress.Output) (io.ReadCloser, int64, error)
+	Download(ctx context.Context, progressOutput progress.Output, cacheDir string) (io.ReadCloser, int64, error)
```

Comment:
Is there a way to avoid having to pass in the `cacheDir` here? This seems problematic that we are passing it here. Shouldn't be associated with the descriptor when it is created or resolved from the cachdir.Manager at that time?